src/main/java/org/ow2/proactive/catalog/util/parser/WorkflowParser.java

```diff
@@ -193,12 +198,24 @@
                 }
             }
 
-            this.keyValueMap = keyValueMapBuilder.build();
-
-            return getKeyValueMap();
+        } catch (XMLStreamException e) {
+            throw new ParsingObjectException(e);
         } finally {
-            xmlStreamReader.close();
+            if (xmlStreamReader != null) {
+                try {
+                    xmlStreamReader.close();
+                } catch (XMLStreamException e) {
+                    throw new ParsingObjectException(e);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not throw exceptions in finally blocks. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1163)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this throw statement from this finally block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1143)